### PR TITLE
OpenShift resource metrics collection and pull and push to OpenStack Object Store

### PR DESCRIPTION
**Pull request by dburugupalli**
Python Flask web application to collect metrics from OpenShift and Push to OpenStack object store. 
at regular intervals

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+SSMT
```
Comment (gagansk) [inline Readme.md]: Please provide a brief description of what this PR is about, why is it necessary, and how to test this PR.

File: api/openshift_info.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import os
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+from helper import helper
+
+
+class ServiceClass(object):
+    def __init__(self):
```
Comment (gagansk) [inline api/openshift_info.py:12]: Please add comments for every function. It should have 3 lines.
1. Purpose 
2. Inputs if any
3. Outputs

File: api/openshift_info.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        pods_failed={}
+
+        for pod in pods.items:
+            if(pod.status.phase == 'Running'):
+                pod_namespace = pod.metadata.namespace
+                pods_running[pod_namespace] =  pods_running.get(pod_namespace, 0) + 1
+            elif(pod.status.phase == 'Failed'):
+                pod_namespace = pod.metadata.namespace
+                pods_failed[pod_namespace] = pods_succeeded.get(pod_namespace, 0) + 1
+            elif(pod.status.phase == 'Succeeded'):
```
Comment (gagansk) [inline api/openshift_info.py:134]: You need to come up with a better conditional statement. Make 'Failed' as the else part.

File: api/openshift_info.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+from helper import helper
+
+
+class ServiceClass(object):
+    def __init__(self):
+    # to run inside the openshift cluster
+        if "OPENSHIFT_BUILD_NAME" in os.environ:
+            service_account_path = '/var/run/secrets/kubernetes.io/serviceaccount'
```
Comment (gagansk) [inline api/openshift_info.py]: Don't use Magic values. Always, assign values to variables and use those variables in the code.

File: app.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+from flask.json import jsonify
+import signal
+import os
+import sys
+
+app = Flask(__name__)
+
+openshift_object = openshift_info.ServiceClass()
+
+@app.route('/health')
```
Comment (gagansk) [inline app.py]: Why are we keeping this? It is not checking for anything.

**Review by gagansk — CHANGES_REQUESTED** [review]
Great job, Sai! 
I have requested some changes. Also, please provide comments for the functions.

File: Readme.md (hunk 1 above)
Comment (dburugupalli) [reply Readme.md]: I updated readme.md and also added few steps to test this PR

File: api/openshift_info.py (hunk 2 above)
Comment (dburugupalli) [reply api/openshift_info.py:12]: I commented the code as following for all the files 
Every Comment Contains 
Purpose: <Purpose>
Input: <Input parameters>
Output: <Output>

File: api/openshift_info.py (hunk 4 above)
Comment (dburugupalli) [reply api/openshift_info.py]: I missed out with this one, I have this code such that it accepts code from environment variables

File: app.py (hunk 5 above)
Comment (dburugupalli) [reply app.py]: As of now, i have removed this

**Comment by dburugupalli** [comment]
Closing this Pull Request. As backend architechture is built on top of metering operator.

**Pull request closed without merging**